homeassistant/components/media_player/cast.py

```diff
@@ -236,7 +237,7 @@ def _async_create_cast_device(hass: HomeAssistantType,
                               port=discovery_info['port'])
     elif CONF_HOST in config:
         info = ChromecastInfo(host=config[CONF_HOST],
-                              port=DEFAULT_PORT)
+                              port=config.get(CONF_PORT, DEFAULT_PORT))
```

Comment:
The default value will have been populated by the platform schema.

Reply:
I'm pretty sure, if I remember well, it isn't just the test but the platform itself.
Not supplying a hardcoded fallback raises a KeyError on 'port'.
This is caught by the hardcoded `DEFAULT_PORT` in the original code, and also illustrated by how `ignore_cec` is handled:

https://github.com/home-assistant/home-assistant/blob/a1f08aecc5ae12e9d5e0e5e152da18e14c7c1708/homeassistant/components/media_player/cast.py#L230

Why the default values are not filled in, I don't know.

Reply:
When is the `KeyError` raised? Normal set up will validate the platform config and make sure defaults are in place.

Reply:
So I'm 99,9% sure, I have had this KeyError raised within HA in my initial tries. Can't get it to trigger anymore, however.

The unittests still fail on this KeyError. I've been looking at it for quite some time but I can't figure out what to do to solve it. I can see it has something to do with passing in the config entries "statically", but I'm not sure how to solve it; I'm too inexperienced for that...

Passing in the 'port' value statically on the affected setups ofcourse solves it, but I guess it's not really a worthy test that way.
It's not that I want to get off easily, it's just that I can't figure out how to properly solve it. If you could point me in the right direction with a concrete pointer, I'd be happy to take another crack at it; otherwise, I'm afraid I'll have to throw the towel in the ring...

Reply:
This is the most simple example test I could find that uses pytest functions and sets up a component with a platform.

https://github.com/home-assistant/home-assistant/blob/dev/tests/components/switch/test_unifi.py#L215-L222

So for each test we need to set up the correct component (media_player) and pass the component config dictionary including the correct platform config.

Then we assert something, most often a state in the state machine.

The existing tests use a helper function to set up the platform `async_setup_cast`. I would first see if we can modify that function to set up the component and platform as described above. Some things in the tests will have to change since we can't mock and pass `async_add_entities` to `async_setup_component`.